fix(curriculum): improve testing in regex challenge

nayabatir1 opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [ ] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

Closes #48225 

<!-- Feel free to add any additional description of changes below this line -->

I've updated the solution regex.

Provided solution `/(Franklin|Eleanor) ([A-Z]\.? )?Roosevelt/`
Eleanor asd. Roosevelt -> FAILS
Eleanor AFASDF. adfcx Roosevelt -> FAILS

nayabatir1 committed:
fix(curriculum): add 2 test cases and updated solution

nayabatir1 committed:
fix(curriculum): solution regex updated

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/check-for-mixed-grouping-of-characters.md:
````diff
@@ -85,6 +99,6 @@ let result = false; // Change this line
 ```js
 let myString = "Eleanor Roosevelt";
-let myRegex = /(Franklin|Eleanor).*Roosevelt/;
+let myRegex = /(Franklin|Eleanor)\s{1}(\S+\s)*Roosevelt/;
````
This regex would pass `Eleanor ?dsg Roosevelt` as valid.  That is a problem.

How about:
```
/(Franklin|Eleanor) (([A-Z]\.?|[A-Z][a-z]+) )?Roosevelt/
```

nayabatir1 replied:
Your regex is failing in this case
![image](https://user-images.githubusercontent.com/91016903/198698196-c7189e6e-dd79-49ba-9dc8-a59b760e4e9d.png)

So I've modified my regex. Shall I use this ?
`(Franklin|Eleanor)\s([A-Za-z]+\.?\s)*Roosevelt`

RandellDawson replied:
For your new regex, `Eleanor dsg. asdf Roosevelt` should fail anyway as it is only supposed to allow for a middle name and not two and definitely not `dsg.`  A valid abbreviation would be a single capital letter followed by a period like 'D.`

RandellDawson reviewed (changes requested):
See my concern and suggestion.

ieahleen commented:
Please take a moment and update the title of the PR so that it follows the [contributing guidelines](https://contribute.freecodecamp.org/)

nayabatir1 commented:
@ieahleen my pr includes two test cases and modified regex exp of the solution.
If you want to include what test cases I've added, the title is going to be way too long.

naomi-lgbt commented:
An appropriate PR title could be something like `fix(curriculum): improve testing in regex challenge`.

nayabatir1 committed:
fix(curriculum): regex expression updated in regex challenge

nayabatir1 committed:
fix(curriculum): solution regex expression updated in regex challenge

naomi-lgbt reviewed (approved):
I think this should be good now?

moT01 reviewed (approved):
LGTM 🎉

What do you think @RandellDawson?

For future reference @nayabatir1, make sure to complete the checklist in the PR description please.

RandellDawson reviewed (approved):
I think these new changes have it covered better now.

naomi-lgbt commented:
Thank you for your contribution to the page! 👍
We are happy to accept these changes and look forward to future contributions. 🎉
